docs/content/latest/explore/ysql-language-features/cursor.md

```diff
@@ -0,0 +1,271 @@
+
+### How to Fetch Rows
+
+Using the `FETCH` statement, you can obtain all rows or a specific row from the cursor and place it into a target such as a record, a row variable, or a comma-separated list of variables. When the rows are exhausted, the target is set to`NULL`.
+
+The following examples show how to apply the `FETCH` statement to a cursor:
+
+```sql
+FETCH ALL FROM employees_cursor_2;
+```
+
+```sql
+FETCH FROM employees_cursor_2 INTO employees_row;
+```
+
+<!--
+
+### How to Fetch Rows
+
+Using the `FETCH` statement, you can obtain the next row from the cursor and place it into a target such as a record, a row variable, or a comma-separated list of variables. When the rows are exhausted, the target is set to`NULL`.
+
+The next row is fetched by default. Using YSQL, you can specify one of the following fetch directions to override `NEXT`:
+
+- `LAST`
+- `PRIOR`
+- `FIRST`
+- `ABSOLUTE count`
+- `RELATIVE count`
+- `FORWARD` (for cursors declared with a `SCROLL` option)
+- `BACKWARD` (for cursors declared with a `SCROLL` option)
+
+The following examples show how to fetch a cursor:
+
+```sql
+FETCH ALL FROM employees_cursor_2;
+```
+
+```sql
+FETCH employees_cursor_2 INTO employees_row;
+```
+
+```sql
+FETCH LAST FROM employees_row INTO employee_no, name, department;
+```
+-->
+
+<!--
+
+### How to Move a Cursor
+
+NOT SUPPORTED ??
+
+The `MOVE` statement allows you to shift the position of the cursor without retrieving any row. You can direct the shift by using the same values as you would use with the [`FETCH` statement](#how-to-fetch-a-row).
+
+The following examples show how to move a cursor:
+
+```sql
+MOVE employees_cursor_2;
+```
+
+```sql
+MOVE forward 3 FROM employees_cursor_1;
+```
+
+-->
+
+<!--
+
+### How to Delete and Update Rows
+
+A table with a cursor positioned on it can be updated or deleted using the cursor as the row identifyer with a `DELETE WHERE CURRENT OF` or `UPDATE WHERE CURRENT OF` statement.
+
+The following example shows how to update a row:
+
+```sql
+UPDATE employees SET department = section 
+  WHERE CURRENT OF employees_cursor_1;
+```
+
+THIS IS SUPPORTED
+
+```sql
+DELETE FROM employees_cursor_2;
+```
+
+-->
+
+### How to Delete Rows
+
+The following example shows how to delete all rows from a cursor:
+
+```sql
+DELETE FROM employees_cursor_2;
+```
+
+### How to Return a Cursor
+
+You can return a cursor using a function that opens the cursor and returns its name to the caller which, in turn, can fetch rows from the cursor and close it before the end of the transaction, if needed.
+
+For more information and examples, refer to the "Returning Cursors" section in [Using Cursors](https://www.postgresql.org/docs/11/plpgsql-cursors.html#PLPGSQL-CURSOR-USING). 
+
+### How to Use Loops
+
+You can iterate through the result set of a bound cursor using a certain form of the `FOR` statement, as per the following syntax:
+
+```
```

Comment:
```suggestion
```sql
```

Reply:
Cursors are a PL/pgSQL language feature. Look [HERE](https://www.postgresql.org/docs/11/plpgsql-cursors.html).

Calling the YSQL cursors is unhelpful. Arguably, PL/pgSQL (and stored procs in general) are part of the YSQL subsystem. But it’s far better to talk about cursors as I said: as a PL/pgSQL language feature.

Your doc must show complete, runnable PL/pgSQL. At least an anonymous block. Ideally a procedure.

One of the most useful cases is for dynamic SQL: opening a cursor for a statement whose text is presented as a `text` variable. You must show an example of this.

Re « The following commands… are associated with YSQL cursors » These are not commands. They’re _statements_ in the PL/pgSQL language.

Search out « bounded ». It should be « bound ».

Reply:
Thanks, I fixed a couple of things. 
WRT a full example, I'll create a separate enhancement ticket for that.

Reply:
Create ticket https://github.com/yugabyte/yugabyte-db/issues/8025 to add example